lib/ex_webrtc/recorder.ex

```diff
@@ -0,0 +1,193 @@
+defmodule ExWebRTC.Recorder do
+  @moduledoc """
+  Saves received RTP packets to a file for later processing/analysis.
+
+  Dumps raw RTP packets fed to it in a custom format. Use `Recorder.Converter` to process them.
+  """
+
+  use GenServer
+
+  alias ExWebRTC.MediaStreamTrack
+
+  require Logger
+
+  @default_base_dir "./recordings"
+
+  @typedoc """
+  Options that can be passed to `start_link/1`.
+
+  * `base_dir` - Base directory where Recorder will save its artifacts. `#{@default_base_dir}` by default.
+  * `on_start` - Callback that will be executed just after the Recorder is (re)started.
+                 It should return the initial list of tracks to be added.
+  """
+  @type option ::
+          {:base_dir, String.t()}
+          | {:on_start, (-> [MediaStreamTrack.t()])}
+
+  @type options :: [option()]
+
+  # Necessary to start Recorder under a supervisor using `{Recorder, [recorder_opts, gen_server_opts]}`
+  @doc false
+  @spec child_spec(list()) :: Supervisor.child_spec()
+  def child_spec(args) do
+    %{
+      id: __MODULE__,
+      start: {__MODULE__, :start_link, args}
+    }
+  end
+
+  @doc """
+  Starts a new `ExWebRTC.Recorder` process.
+
+  `ExWebRTC.Recorder` is a `GenServer` under the hood, thus this function allows for
+  passing the generic `t:GenServer.options/0` as an argument.
+  """
+  @spec start(options(), GenServer.options()) :: GenServer.on_start()
+  def start(recorder_opts \\ [], gen_server_opts \\ []) do
+    GenServer.start(__MODULE__, recorder_opts, gen_server_opts)
+  end
+
+  @doc """
+  Starts a new `ExWebRTC.Recorder` process.
+
+  Works identically to `start/2`, but links to the calling process.
+  """
+  @spec start_link(options(), GenServer.options()) :: GenServer.on_start()
+  def start_link(recorder_opts \\ [], gen_server_opts \\ []) do
+    GenServer.start_link(__MODULE__, recorder_opts, gen_server_opts)
+  end
+
+  @doc """
+  Adds new tracks to the recording.
+  """
+  @spec add_tracks(GenServer.server(), [MediaStreamTrack.t()]) :: :ok | :error
+  def add_tracks(recorder, tracks) do
+    # XXX need?
+    try do
+      GenServer.call(recorder, {:add_tracks, tracks})
+    catch
+      _exit_or_error, _e ->
+        Logger.error("Recorder is down, not adding tracks")
+        :error
+    end
```

Comment:
Up for discussion whether this `try` block should go here

I don't think this is the correct place for it, I'd put it in Broadcaster logic instead